drivers/modem/gsm_ppp.c

```diff
@@ -587,7 +586,8 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 		goto attaching;
 	}
 
-	if (IS_ENABLED(CONFIG_GSM_MUX) && gsm->mux_enabled) {
```

Comment:
Why this change? We typically tend to include `IS_ENABLED` checks within C code instead of preprocessor when possible - it gives better code coverage during compilation. Same would go for some other places below.